lib/datapacksjob.js

```diff
@@ -3082,24 +3082,28 @@ DataPacksJob.prototype.downloadPerformanceData = async function (jobInfo) {
                     "remoteOptions.matrixName", "Type", "Sub Type", "Language", "preIP", "postIP", "remoteClass", "remoteMethod", "returnFullDataJSON"];
 
                 while (unique_omni_id.length > 0) {
+                    try{
+                        let chunk = unique_omni_id.splice(0, 150);
+                        let unique_omni_id_format = "";
 
-                    let chunk = unique_omni_id.splice(0, 150);
-                    let unique_omni_id_format = "";
+                        chunk.forEach(omniId => {
+                            if (omniId) {
+                                unique_omni_id_format += `'${omniId}',`;
+                            }
+                        });
+                        if (unique_omni_id_format) {
```

Comment:
Wait actually... will this while loop actually finish?

Reply:
no... try and catch is inside while loop